audio/codec2/Portfile

```diff
@@ -22,12 +22,11 @@ long_description    Codec 2 is an open source speech codec designed for \
     designed for digital voice over HF radio.
 homepage            http://www.rowetel.com/codec2.html
 
-# migrated from https://svn.code.sf.net/p/freetel/code/codec2/tags/${version}
-github.setup        drowe67 codec2 ff5841a18bfd9df0e8a250dc57fb7388cabccda1
-version             0.8.1
-checksums           rmd160  0e4c0d91e893b4e20e609be77fca037c246999c5 \
-                    sha256  2aa2db1abb7ef4878341e4858deb02a795fb136e3692eb35a5bd677f2aedecc9 \
-                    size    12272696
+github.setup        drowe67 codec2 567346818c0d4d697773cf66d925fdb031e15668
+version             20190819-[string range ${github.version} 0 7]
```

Comment:
I am not a fan of including part of the git commit version in the macports version string, as there is no guarantee they will always increase monotonically. Say, hypothetically speaking, you did make two releases on the same day for some reason, for two different tags. There is no garantee the second would be seen as an update to the first.

I would suggest removing this from the version, and just use the date. If you did, unlikely as it probably is, need to make two releases in the same day, just increase the `revision` for it.

Reply:
You are right and we discussed in the past with @michaelld and we choose this direction to expose easily the commit to the user (no tags on these ports). We are conscious about the monotonically fault and in fact if that happen we use `revision` as bump.

Reply:
bumping the revision does not help if you keep the github tag in the version. e.g. ay you release

`20180820-0000002`

then later in the day

`20180820-0000001`

which just so happens to be a newer commit. bumping the revision on that last version does not help, it will still be considered older than the first. The only option is to bump the epoch, which is never a nice thing.

Reply:
I remembered from personal tests that it works but seems not as you said (now I checked `macports.tcl`).
Obviously epoch should be avoided seen that is permanent.
Anyway, change two commits, with the last smaller than the first, on the same day after personal tests is quite improbable and personally I like to see immediately the commit.
let's hear @michaelld say.

Reply:
changing `revision` only makes a different if `version` is unchanged. If `version` changes, that always trumps any `revision` changes.

Reply:
I see that behaviour on `macports.tcl`. thanks for point out the problem!

Reply:
p.s. I agree its an unlikely scenario. But the loop hole is there and in my opinion easy to avoid by just not putting the github commit id into the version. I fail to see what real advantage there is in exposing this info to the user, as you say above.

Reply:
I can speak for myself:
if I do for example `port info codec2` or/and `port installed codec2` I can check immediately in github which changes the software implements and if there are some troubles without opening the `Portfile`. Maybe there is another method that I don't know.
Note: generally this version is used only on `-devel` ports

Reply:
Ok. As I have said, I am not a fan, but won’t object to it being merged as is.

Reply:
Hitting this discussion a little post-facto LOL ... I feel what you mention is a reasonable tradeoff to make: remove the burden from the end-user for tracking down the exact version of some specific port being built/installed/used, and put it onto the port developer via this version string in the sense of making sure that the version increases monotonically -- which as you note can be tricky. I'm willing to make that tradeoff, and will do so to the best of my ability for my ports.

Reply:
sorry @michaelld I didn't understand the new "line". thanks